File: SharedApi/Models/Transaction.cs
using System;
using System.Collections.Generic;
using SharedLibrary.Enums;

namespace SharedApi.Models
{
	public partial class Transaction
	{
		public int ID { get; set; }
		public int FromID { get; set; }
		public int ToID { get; set; }
		public int ZolikID { get; set; }
		public System.DateTime Date { get; set; }
		public string Message { get; set; }
		public TransactionAssignment Typ { get; set; }
	}
}

File: MobileApp/zoliky_teachers/lib/utils/api/models/Transaction.dart
import 'package:zoliky_teachers/utils/Global.dart';
import 'package:zoliky_teachers/utils/api/enums/TransactionAssignment.dart';
import 'package:zoliky_teachers/utils/api/models/ZolikType.dart';
import 'package:zoliky_teachers/utils/interfaces/IDbObject.dart';

class Transaction implements IDbObject {
  int id;
  int fromID;
  int toId;
  int zolikId;
  int zolikTypeId;
  String from;
  String to;
  DateTime date;
  String message;
  TransactionAssignment type;
  String zolikTitle;

  ZolikType get zolikType => Global.getTypeById(this.zolikTypeId);

  Transaction();

  Transaction.fromJson(Map<String, dynamic> json)
      : id = json["ID"],
        fromID = json["FromID"],
        toId = json["ToID"],
        zolikId = json["ZolikID"],
        zolikTypeId = json["ZolikTypeID"],
        from = json["From"],
        to = json["To"],
        date = DateTime.parse(json["Date"]),
        message = json["Message"],
        type = TransactionAssignment.fromId(json["Typ"]),
        zolikTitle = json["ZolikTitle"];
}
